File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/models/FeedBacks/AlumniFeedBackonCompany.cs
using PlacementCellBackend.Models.Enums;
using PlacementCellBackend.Models.InterviewRounds;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using PlacementCellBackend.Models;

namespace  PlacementCellBackend.Models.FeedBacks;

public class AlumniFeedBackonCompany
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public string Id { get; set; } = string.Empty;

    // College reference - feedback is college-specific
    [Required]
    public int CollegeId { get; set; }

    [ForeignKey("CollegeId")]
    public College? College { get; set; }

    [Required]
    public string CompanyId { get; set; } = string.Empty;

    [ForeignKey("CompanyId")]
    public Company? Company { get; set; }

    [Required]
    public string AlumniId { get; set; } = string.Empty;

    [ForeignKey("Id")]
    public Alumni? Alumni { get; set; }

    [Required]
    public string JobProfile { get; set; } = string.Empty;

    [Required]
    public JobType JobType { get; set; }

    [Required]
    public string JobLocation { get; set; } = string.Empty;

    [Required]
    public string CTC { get; set; } = string.Empty;

    [Required]
    public WorkMode WorkMode { get; set; }

    public CodingRound? CodingRoundInfo { get; set; }
    public TechnicalRound? TechnicalRoundInfo { get; set; }
    public HRRound? HRRoundInfo { get; set; }
    public Resources? ResourcesInfo { get; set; }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/models/FeedBacks/AlumniFeedBackonCompany.java
package com.example.placementicsbackend.models.FeedBacks;

import com.example.placementicsbackend.models.Alumni;
import com.example.placementicsbackend.models.Company;
import com.example.placementicsbackend.models.Enums.JobType;
import com.example.placementicsbackend.models.Enums.WorkMode;
import com.example.placementicsbackend.models.InterviewRounds.CodingRound;
import com.example.placementicsbackend.models.InterviewRounds.HRRound;
import com.example.placementicsbackend.models.InterviewRounds.Resources;
import com.example.placementicsbackend.models.InterviewRounds.TechnicalRound;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.MapsId;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "alumnifeedbackoncompany")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AlumniFeedBackonCompany {

    @Id
    @Column(name = "Id")
    private String id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "CompanyId", referencedColumnName = "CompanyId")
    private Company company;

    @Column(name = "AlumniId", nullable = false)
    private String alumniId;

    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @MapsId
    @JoinColumn(name = "Id", referencedColumnName = "Id")
    private Alumni alumni;

    @Column(name = "JobProfile", nullable = false)
    private String jobProfile = "";

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "JobType", nullable = false)
    private JobType jobType;

    @Column(name = "JobLocation", nullable = false)
    private String jobLocation = "";

    @Column(name = "CTC", nullable = false)
    private String ctc = "";

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "WorkMode", nullable = false)
    private WorkMode workMode;

    @Embedded
    private CodingRound codingRoundInfo;

    @Embedded
    private TechnicalRound technicalRoundInfo;

    @Embedded
    private HRRound hrRoundInfo;

    @Embedded
    private Resources resourcesInfo;
}
